### ISSUE TITLE
Imageio is unable to read meta data from avconf output

### ISSUE THREAD

**Issue by cancan101** [imageio/imageio#171]
Using this mp4: https://www.dropbox.com/s/vvou73bjf27t2ds/black-2.mp4?dl=0 and `avconv`, when I try to read I get:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/imageio/core/functions.py", line 111, in get_reader
    return format.get_reader(request)
  File "/usr/local/lib/python2.7/site-packages/imageio/core/format.py", line 158, in get_reader
    return self.Reader(self, request)
  File "/usr/local/lib/python2.7/site-packages/imageio/core/format.py", line 207, in __init__
    self._open(**self.request.kwargs.copy())
  File "/usr/local/lib/python2.7/site-packages/imageio/plugins/ffmpeg.py", line 287, in _open
    self._load_infos()
  File "/usr/local/lib/python2.7/site-packages/imageio/plugins/ffmpeg.py", line 463, in _load_infos
    fps = float(line[match.start():match.end()].split(' ')[1])
AttributeError: 'NoneType' object has no attribute 'start'
```

**Comment by almarklein** [comment]
This suggest that my suspicion in #170 is correct. We'd need to add a second path to parse the meta data for avconv, or make the parsing more robust so it works for avconv too.

Hard to debug from my end, though if you can copy the printout from avconf that imageio is trying to parse, I might be able to fix it.

**Comment by almarklein** [comment]
I can spin up a VM, install avconv etc., but I am low on cycles and a text dump of the header that needs to be parsed would really help.

Also cc-ing @kuchi on this.

**Comment by kuchi** [comment]
Ok, so here is the output from ffmpeg versus avconv:

**FFMPEG output:**
```
ffmpeg -i black-2.mp4
ffmpeg version 3.2.2 Copyright (c) 2000-2016 the FFmpeg developers
  built with Apple LLVM version 8.0.0 (clang-800.0.42.1)
  configuration: --prefix=/usr/local/Cellar/ffmpeg/3.2.2 --enable-shared --enable-pthreads --enable-gpl --enable-version3 --enable-hardcoded-tables --enable-avresample --cc=clang --host-cflags= --host-ldflags= --enable-libmp3lame --enable-libx264 --enable-libxvid --enable-opencl --disable-lzma --enable-vda
  libavutil      55. 34.100 / 55. 34.100
  libavcodec     57. 64.101 / 57. 64.101
  libavformat    57. 56.100 / 57. 56.100
  libavdevice    57.  1.100 / 57.  1.100
  libavfilter     6. 65.100 /  6. 65.100
  libavresample   3.  1.  0 /  3.  1.  0
  libswscale      4.  2.100 /  4.  2.100
  libswresample   2.  3.100 /  2.  3.100
  libpostproc    54.  1.100 / 54.  1.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'black-2.mp4':
  Metadata:
    major_brand     : mp42
    minor_version   : 1
    compatible_brands: mp41mp42isom
    creation_time   : 2016-10-17T23:52:49.000000Z
  Duration: 00:00:00.10, start: 0.000000, bitrate: 105 kb/s
    Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p(tv), 375x491 [SAR 1:1 DAR 375:491], 47 kb/s, 19.35 fps, 19.35 tbr, 600 tbn, 1200 tbc (default)
    Metadata:
      creation_time   : 2016-10-17T23:52:49.000000Z
      handler_name    : Core Media Video
```

**AVCONV output**

```
avconv -i black-2.mp4
avconv version 11.4, Copyright (c) 2000-2014 the Libav developers
  built on Sep 13 2016 05:26:47 with Apple LLVM version 8.0.0 (clang-800.0.38)
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'black-2.mp4':
  Metadata:
    major_brand     : mp42
    minor_version   : 1
    compatible_brands: mp41mp42isom
    creation_time   : 2016-10-17 23:52:49
  Duration: 00:00:00.10, start: 0.000000, bitrate: 105 kb/s
    Stream #0.0(und): Video: h264 (High), yuv420p, 376x492 [PAR 1:1 DAR 94:123], 47 kb/s, 600 tbn (default)
    Metadata:
      creation_time   : 2016-10-17 23:52:49
```


so the main issue I see is that `avconv` is missing the fps for this file but ffmpeg does print the fps. However, for other video files, avconv does show the fps, for example on a different file the output is:

```
avconv -i small.mp4
avconv version 11.4, Copyright (c) 2000-2014 the Libav developers
  built on Sep 13 2016 05:26:47 with Apple LLVM version 8.0.0 (clang-800.0.38)
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'small.mp4':
  Metadata:
    major_brand     : mp42
    minor_version   : 0
    compatible_brands: mp42isomavc1
    creation_time   : 2010-03-20 21:29:11
    encoder         : HandBrake 0.9.4 2009112300
  Duration: 00:00:05.56, start: 0.000000, bitrate: 551 kb/s
    Stream #0.0(und): Video: h264 (Constrained Baseline), yuv420p, 560x320, 465 kb/s, 30 fps, 90k tbn, 60 tbc (default)
    Metadata:
      creation_time   : 2010-03-20 21:29:11
      encoder         : JVT/AVC Coding
    Stream #0.1(eng): Audio: aac, 48000 Hz, mono, fltp, 83 kb/s (default)
    Metadata:
      creation_time   : 2010-03-20 21:29:11
```

So I don't think this is a bug in imageio, it's just a problem with this video file and perhaps a reason to use ffmpeg since it seems to be more robust (well at least in this case).

@cancan101, I don't really use avconv, are there some cases where it is better than ffmpeg?

**Comment by cancan101** [comment]
A number of Ubuntu versions only had avconv in the repositories.

**Comment by kuchi** [comment]
doesn't imagio come with it's own ffmpeg that it downloads? I know the whole avconv / ffmpeg thing is really annoying. 

Did you generate the black video? I'm just curious why it is not getting all the meta data read correctly in avconv. If you did make it, how did you generate it?

**Comment by almarklein** [comment]
I verified that the other needed meta info can be read. We can set fps to 0, and nframes to inf, which still allows reading the frames. Implemented in #205
